src/Simulation/CsharpGeneration.Tests/Circuits/EntryPointTests.qs

```diff
@@ -0,0 +1,247 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+//
+// No Options
+//
+
+namespace Microsoft.Quantum.CsharpGeneration.Testing.EntryPoint {
+    @EntryPoint()
+    operation ReturnUnit() : Unit { }
+}
+
+// ---
+
+namespace Microsoft.Quantum.CsharpGeneration.Testing.EntryPoint {
+    @EntryPoint()
+    operation ReturnInt() : Int {
+        return 42;
+    }
+}
+
+// ---
+
+namespace Microsoft.Quantum.CsharpGeneration.Testing.EntryPoint {
+    @EntryPoint()
+    operation ReturnString() : String {
+        return "Hello, World!";
+    }
+}
+
+// ---
+
+//
+// Single Option
+//
+
+namespace Microsoft.Quantum.CsharpGeneration.Testing.EntryPoint {
+    @EntryPoint()
+    operation AcceptInt(n : Int) : Int {
+        return n;
+    }
+}
+
+// ---
+
+namespace Microsoft.Quantum.CsharpGeneration.Testing.EntryPoint {
+    @EntryPoint()
+    operation AcceptBigInt(n : BigInt) : BigInt {
+        return n;
+    }
+}
+
+// ---
+
+namespace Microsoft.Quantum.CsharpGeneration.Testing.EntryPoint {
+    @EntryPoint()
+    operation AcceptDouble(n : Double) : Double {
+        return n;
+    }
+}
+
+// ---
+
+namespace Microsoft.Quantum.CsharpGeneration.Testing.EntryPoint {
+    @EntryPoint()
+    operation AcceptBool(b : Bool) : Bool {
+        return b;
+    }
+}
+
+// ---
+
+namespace Microsoft.Quantum.CsharpGeneration.Testing.EntryPoint {
+    @EntryPoint()
+    operation AcceptPauli(p : Pauli) : Pauli {
+        return p;
+    }
+}
+
+// ---
+
+namespace Microsoft.Quantum.CsharpGeneration.Testing.EntryPoint {
+    @EntryPoint()
+    operation AcceptResult(r : Result) : Result {
+        return r;
+    }
+}
+
+// ---
+
+namespace Microsoft.Quantum.CsharpGeneration.Testing.EntryPoint {
+    @EntryPoint()
+    operation AcceptRange(r : Range) : Range {
+        return r;
+    }
+}
+
+// ---
+
+namespace Microsoft.Quantum.CsharpGeneration.Testing.EntryPoint {
+    @EntryPoint()
+    operation AcceptString(s : String) : String {
+        return s;
+    }
+}
+
+// ---
+
+namespace Microsoft.Quantum.CsharpGeneration.Testing.EntryPoint {
+    @EntryPoint()
+    operation AcceptStringArray(xs : String[]) : String[] {
```

Comment:
Does something like a Range[] as argument work fine? Looking at the code I think it should, but wouldn't hurt to check.

Reply:
Thanks for bringing this up, I didn't consider it. No, it doesn't work, for the same reason that nested arrays don't work - since range arguments can be space-separated, it doesn't know when one range stops and the other one begins.

We could either require ranges to not have spaces (only allow `1..2..10`, not `1 2 10`), or disallow range arrays. I like the first option (since it's Q# syntax anyway).

Reply:
There is actually a larger issue that is preventing this from being fixed. Custom argument handlers are currently not used for arrays, so arrays of `BigInt`s and `Result`s are not working either.

Reply:
As discussed, that is fine. Sooner or later we should look into supporting this, but this is a good first version. Thanks!